src/emulator/auth/operations.ts

```diff
@@ -2221,7 +2221,14 @@ async function mfaSignInFinalize(
 
   let { user, signInProvider } = parsePendingCredential(state, reqBody.mfaPendingCredential);
   const enrollment = user.mfaInfo?.find(
-    (enrollment) => enrollment.unobfuscatedPhoneInfo === phoneNumber,
+    (enrollment) => {
+      return
+        // all but firebase-ios-sdk finalize with unobfuscated phone number which we find easily...
+        enrollment.unobfuscatedPhoneInfo === phoneNumber ||
+        // ...but firebase-ios-sdk finalizes with obfuscated
+        // This works against cloud auth, so emulator should attempt to find enrollment obfuscated as well.
+        ('+********' + enrollment.unobfuscatedPhoneInfo.substring(enrollment.unobfuscatedPhoneInfo.length - 4)) === phoneNumber;
```

Comment:
The code review bot makes a reasonable point.

My response / updated patch would be more verbose as suggested for readability, and re-use the existing `obfuscatePhoneNumber` method I did not realize existed.

For your consideration:

```suggestion
        // All but firebase-ios-sdk finalize with unobfuscated phone number.
        if (enrollment.unobfuscatedPhoneInfo === phoneNumber) {
          return true;
        }
        
        // But firebase-ios-sdk finalizes with an obfuscated number. This works against
        // cloud auth, so emulator should attempt to find enrollment obfuscated as well.
        if (obfuscatePhoneNumber(enrollment.unobfuscatedPhoneInfo === phoneNumber) {
          return true;
        }
        
        return false;
```